.agents/skills/aicr-uat-report/uat_report.py

```
@@ -108,13 +108,31 @@ def failed_steps(repo, run_url):
+
+    The caller filters for expiry so it can distinguish "never uploaded"
+    (infra failure before prep) from "aged out of retention" — different
+    classifications that a silent filter here would collapse into one.
+    """
     arts = gh_json(
         ["api", f"repos/{repo}/actions/runs/{run_id}/artifacts", "--jq", ".artifacts"]
     )
     return [a for a in arts if DEBUG_ARTIFACT_HINT in a.get("name", "")]
 
 
+def slug(*parts):
```

Ruff (0.15.21)

[warning] Missing type annotation for `*parts`

(ANN002)